### ISSUE TITLE
[JENKINS-67894] Prevent Windows installer from using Java 8

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/packaging#695]
<p><del>The Windows installer for Jenkins should recommend Java 11, especially if it detects Java 8 as the current JVM.</del> (Split out into <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a>.)</p>

<p>When Jenkins requires Java 11 or newer, the Windows installer should prevent installation if it detects that the Jenkins war file will be run with Java 8.  When Java 11 is required, attempts to run with Java 8 should be expected to fail.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894" target="_blank">Prevent Windows installer from using Java 8</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/slide_o_mix.png" title="slide_o_mix's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slide_o_mix">slide_o_mix</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: packaging
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2022-08-18T01:22:14+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-26
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><del>The Windows installer for Jenkins should recommend Java 11, especially if it detects Java 8 as the current JVM.</del> (Split out into <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a>.)</p>

<p>When Jenkins requires Java 11 or newer, the Windows installer should prevent installation if it detects that the Jenkins war file will be run with Java 8.  When Java 11 is required, attempts to run with Java 8 should be expected to fail.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Windows MSI installer for Jenkins</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/timja.png" title="timja's avatar" /> timja:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=422338&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-422338">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Best tracked on GitHub if there's still something you think should be done</p>
</pre>
</details>

</ul>
</details>

<p>Best tracked on GitHub if there's still something you think should be done</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/slide_o_mix.png" title="slide_o_mix's avatar" /> slide_o_mix:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=422434&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-422434">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have the change ready (I won't do a PR yet), just let me know when we are ready for a PR</p>
</pre>
</details>

</ul>
</details>

<p>I have the change ready (I won't do a PR yet), just let me know when we are ready for a PR</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=423906&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423906">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This ticket as filed by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> seems to contain two separate tasks:
 </p>
<ol>
	<li>The Windows installer for Jenkins should recommend Java 11, especially if it detects Java 8 as the current JVM.</li>
	<li>When Jenkins requires Java 11 or newer, the Windows installer should prevent installation if it detects that the Jenkins war file will be run with Java 8. When Java 11 is required, attempts to run with Java 8 should be expected to fail.</li>
</ol>


<p>It seems that task #1 can be integrated and released today, while task #2 cannot be integrated and released until we decide to drop support for Java 8. Is my understanding correct?</p>

<p>If so, then I would propose that we repurpose this ticket to cover task #2 while filing a separate ticket for task #1 and having <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slide_o_mix" class="user-hover" rel="slide_o_mix">slide_o_mix</a> propose the PR, merge, and release it immediately. Thoughts?</p>
</pre>
</details>

</ul>
</details>

<p>This ticket as filed by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> seems to contain two separate tasks:<br/>
 </p>
<ol>
	<li>The Windows installer for Jenkins should recommend Java 11, especially if it detects Java 8 as the current JVM.</li>
	<li>When Jenkins requires Java 11 or newer, the Windows installer should prevent installation if it detects that the Jenkins war file will be run with Java 8. When Java 11 is required, attempts to run with Java 8 should be expected to fail.</li>
</ol>


<p>It seems that task #1 can be integrated and released today, while task #2 cannot be integrated and released until we decide to drop support for Java 8. Is my understanding correct?</p>

<p>If so, then I would propose that we repurpose this ticket to cover task #2 while filing a separate ticket for task #1 and having <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slide_o_mix" class="user-hover" rel="slide_o_mix">slide_o_mix</a> propose the PR, merge, and release it immediately. Thoughts?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=423909&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423909">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sounds fine to me.</p>
</pre>
</details>

</ul>
</details>

<p>Sounds fine to me.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=423910&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423910">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Split out into <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a>. This ticket needs to remain blocked on dropping support for Java 8. <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a> can be merged and released immediately.</p>
</pre>
</details>

</ul>
</details>

<p>Split out into <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a>. This ticket needs to remain blocked on dropping support for Java 8. <a href="https://issues.jenkins.io/browse/JENKINS-68170" title="Recommend Java 11 in Windows installer" class="issue-link" data-issue-key="JENKINS-68170"><del>JENKINS-68170</del></a> can be merged and released immediately.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=427085&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-427085">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slide_o_mix" class="user-hover" rel="slide_o_mix">slide_o_mix</a> If you are interested, this task can now be worked on, because I just merged the PR to require Java 11 in Jenkins core this morning.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slide_o_mix" class="user-hover" rel="slide_o_mix">slide_o_mix</a> If you are interested, this task can now be worked on, because I just merged the PR to require Java 11 in Jenkins core this morning.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/slide_o_mix.png" title="slide_o_mix's avatar" /> slide_o_mix:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=427905&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-427905">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>See <a href="https://github.com/jenkinsci/packaging/pull/322" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/pull/322</a></p>
</pre>
</details>

</ul>
</details>

<p>See <a href="https://github.com/jenkinsci/packaging/pull/322" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/pull/322</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/slide_o_mix.png" title="slide_o_mix's avatar" /> slide_o_mix:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-67894?focusedId=428364&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-428364">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I believe this should be part of the next weekly?</p>
</pre>
</details>

</ul>
</details>

<p>I believe this should be part of the next weekly?</p>
